cypress/support/generateUserData.js

```diff
@@ -0,0 +1,81 @@
+const { faker } = require('@faker-js/faker');
+
+function generateData() {
+  const firstName = faker.person.firstName();
+  const lastName = faker.person.lastName();
+  const email = faker.internet.email({ firstName });
+  const mobileNumber = faker.phone.number('##########');
+
+  return { firstName, lastName, email, mobileNumber };
+}
+
+const getGenderAccordingToName = (firstName) => {
+  const maleNames = ['John', 'James', 'David', 'Michael', 'Robert', 'Jimmy'];
+  const femaleNames = ['Mary', 'Jennifer', 'Linda', 'Patricia', 'Jessica'];
+
+  if (maleNames.includes(firstName)) {
+    return 'male';
+  }
+
+  if (femaleNames.includes(firstName)) {
+    return 'female';
+  }
+
+  return faker.person.sex();
+};
+
+function generatedDateOfBirth() {
+  const randomDate = faker.date.birthdate({ mode: 'age', min: 16, max: 30 });
+  const date = randomDate.getDay();
```

Comment:
The `getDay()` method returns the day of the week (0-6) rather than the day of the month. You should use `getDate()` to get the day of the month.